trpcultivate_phenotypes/src/Plugin/Validators/DuplicateTraits.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/**
+ * @file
+ * Contains validator plugin definition.
+ */
+
+namespace Drupal\trpcultivate_phenotypes\Plugin\Validators;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\trpcultivate_phenotypes\Service\TripalCultivatePhenotypesTraitsService;
+use Drupal\trpcultivate_phenotypes\TripalCultivatePhenotypesValidatorBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Validate duplicate traits within a file
+ *
+ * @TripalCultivatePhenotypesValidator(
+ *   id = "trpcultivate_phenotypes_validator_duplicate_traits",
+ *   validator_name = @Translation("Duplicate Traits Validator"),
+ *   validator_scope = "FILE ROW",
+ * )
+ */
+class DuplicateTraits extends TripalCultivatePhenotypesValidatorBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * A nested array of already validated values forming the unique trait name +
+   *   method name + unit combinations that have been encountered by this
+   *   validator so far within the input file
```

Comment:
For arrays I always find it really helpful to fully lay out the keys and values. Especially when they are nested!

```suggestion
   *   validator so far within the input file. More specifically,
   *   - TRAIT-NAME: array of methods associated with this trait.
   *     - METHOD-NAME: array of units associated with this trait-method combo
   *       - UNIT-NAME: 1 (indicates this full trait-method-unit combo exists)
   *
   *   NOTE: All capital words are intended to be replaced by the actual full
   *   name of the term.
```